python/ray/data/_internal/plan.py

```diff
@@ -513,22 +515,31 @@ def execute(
         """
         context = DatasetContext.get_current()
         if not ray.available_resources().get("CPU"):
-            logger.get_logger().warning(
-                "Warning: The Ray cluster currently does not have "
-                "any available CPUs. The Dataset job will hang unless more CPUs "
-                "are freed up. A common reason is that cluster resources are "
-                "used by Actors or Tune trials; see the following link "
-                "for more details: "
-                "https://docs.ray.io/en/master/data/dataset-internals.html#datasets-and-tune"  # noqa: E501
-            )
+            if log_once("cpu_warning"):
+                logger.get_logger().warning(
+                    "Warning: The Ray cluster currently does not have "
+                    "any available CPUs. The Dataset job will hang unless more CPUs "
+                    "are freed up. A common reason is that cluster resources are "
+                    "used by Actors or Tune trials; see the following link "
+                    "for more details: "
+                    "https://docs.ray.io/en/master/data/dataset-internals.html#datasets-and-tune"  # noqa: E501
+                )
         if not self.has_computed_output():
             if self._run_with_new_execution_backend():
                 from ray.data._internal.execution.bulk_executor import BulkExecutor
+                from ray.data._internal.execution.streaming_executor import (
+                    StreamingExecutor,
+                )
                 from ray.data._internal.execution.legacy_compat import (
                     execute_to_legacy_block_list,
                 )
 
-                executor = BulkExecutor(copy.deepcopy(context.execution_options))
+                if context.use_streaming_executor:
+                    executor = StreamingExecutor(
```

Comment:
Is this to make all consumptions (not just `iter_batches`) use streaming?

Reply:
Yeah, this should cover the remaining cases